visualization/app/codeCharta/codeCharta.component.ts

```diff
@@ -67,9 +68,7 @@ export class CodeChartaController {
 	}
 
 	public onClick() {
-		if (this.storeService.getState().appSettings.searchPanelMode !== SearchPanelMode.minimized) {
-			this.storeService.dispatch(setSearchPanelMode(SearchPanelMode.minimized))
-		}
+		this.codeChartaMouseEventService.closeComponentExcept()
```

Comment:
This was a complicated issue. We have a global event listener that registers a click to close the search-panel for example, when clicking somewhere else. Now, if we click in the search-bar, the search-bar will open. The event-propagation is stopped to prevent the listener from closing the search-bar again.

Now, we want to do the same with the other panels in this PR. We open one of the metric-panels and then click on the search panel. The click-event won't be propagated and the metric-panel will not close. Instead, both the search-panel and the metric-panel would be open.

To solve this issue, we can just close all panels by dispatching an action (triggering a redux event), but this would close the search-panel we just opened. That's why this function has a list of actions, that would close panels and you can provide an optional function as a parameter that would be triggered from the list. I think we should list those functions in an enum. So for the searchBar, we'd call `this.codeChartaMouseEventService.closeComponentExcept(this.codeChartaMouseEventService.closeSearchPanel)`

Reply:
Ah, in that case maybe name it `closeComponentExceptCurrent` or `closeAllButLastComponent`?

Reply:
`closeComponentExceptCurrent` sounds good. I also renamed the parameter to `currentCloseFunction`. Hopefully, this makes things clearer.